web/src/components/navbar/NavBarDashboardLink.tsx

```diff
@@ -0,0 +1,47 @@
+import { UnStyledButton } from "@/components/njwds-extended/UnStyledButton";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { removeBusiness } from "@/lib/domain-logic/removeBusiness";
+import { ROUTES } from "@/lib/domain-logic/routes";
+import analytics from "@/lib/utils/analytics";
+import { Business } from "@businessnjgovnavigator/shared/userData";
+import { useRouter } from "next/router";
+import { ReactNode } from "react";
+
+interface Props {
+  businessName: string;
```

Comment:
pebble - do we need this as a prop, can we pull it from the `useUserData` hook? Same question with linkText? It's feel a bit confustion why we are using dependency injection here

Reply:
yes we can, I initially wasn't using the `useUserData` hook but I had to in order for the return to previous business to work, but I didn't catch this refactor